google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractReadContext.java

```diff
@@ -389,6 +396,7 @@ void initTransaction() {
   final SpannerRpc rpc;
   final ExecutorProvider executorProvider;
   Span span;
```

Comment:
Idk enough about this code to answer, maybe someone from spanner can. My guess is that the span comes from another thread.

When the code is fully migrated to OTel, passing Context instead of Span would be more appropriate